fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/UnsetVariableCommand.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.doris.common.ErrorCode;
+import org.apache.doris.common.ErrorReport;
+import org.apache.doris.common.UserException;
+import org.apache.doris.mysql.privilege.PrivPredicate;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.StmtExecutor;
+import org.apache.doris.qe.VariableMgr;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * UnSetVarOp
+ */
+public class UnsetVariableCommand extends Command implements Forward {
+    private static final Logger LOG = LogManager.getLogger(StmtExecutor.class);
+
+    private SetType setType;
+
+    // variables to restore
+    private String variable = null;
+
+    private boolean applyToAll = false;
+
+    public UnsetVariableCommand(SetType setType, String varName) {
+        super(PlanType.UNSET_VARIABLE_COMMAND);
+        this.setType = setType;
+        this.variable = varName;
+    }
+
+    public UnsetVariableCommand(SetType setType, boolean applyToAll) {
+        super(PlanType.UNSET_VARIABLE_COMMAND);
+        this.setType = setType;
+        this.applyToAll = applyToAll;
+    }
+
+    public SetType getSetType() {
+        return setType;
+    }
+
+    public String getVariable() {
+        return variable;
+    }
+
+    public boolean isApplyToAll() {
+        return applyToAll;
+    }
+
+    private void validate() throws UserException {
+        if (StringUtils.isEmpty(variable) && !applyToAll) {
+            throw new AnalysisException("You should specific the unset variable.");
+        }
+
+        if (setType == SetType.GLOBAL) {
+            if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+                ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR,
+                        "ADMIN");
+            }
+        }
+    }
+
+    /**
+     * return sql expression of this command
+     * @return string of this command
+     */
+    public String toSql() {
+        StringBuilder sb = new StringBuilder();
+
+        sb.append("UNSET ");
+        sb.append(setType).append(" VARIABLE ");
+        if (!StringUtils.isEmpty(variable)) {
+            sb.append(variable).append(" ");
+        } else if (applyToAll) {
+            sb.append("ALL");
+        }
+        return sb.toString();
+    }
+
+    @Override
+    public RedirectStatus toRedirectStatus() {
+        if (setType == SetType.GLOBAL) {
+            return RedirectStatus.FORWARD_WITH_SYNC;
+        }
+
+        return RedirectStatus.NO_FORWARD;
+    }
+
+    @Override
+    public void afterForwardToMaster(ConnectContext context) throws Exception {
+        if (isApplyToAll()) {
+            VariableMgr.setAllVarsToDefaultValue(context.getSessionVariable(), SetType.SESSION);
```

Comment:
why setType is SESSION?

Reply:
because when follower unset variable in global mode, it would trigger forward_with_sync mode decided by toRedirectStatus and calling forwardToMaster. When forwardToMaster, master would write it to editLog which means it would persist to Env 'storage'. And when follower restart, it replayJournal from last updated one, which is sync with master. So here the only thing it need to do is set current variable in session mode.